src/ot/Transformation.js

```diff
@@ -150,13 +150,31 @@ const Transform = {
     },
     /**
      * [insert_text, add_mark] transformation.
-     * @param {Operation} op1 
-     * @param {Operation} op2 
-     * @param {String} side 
+     * @param {Operation} op1
+     * @param {Operation} op2
+     * @param {String} side
      */
-    // transformInsTextAddMark: (op1, op2, side) => {
-
-    // },
+    transformInsTextAddMark: (op1, op2, side) => {
+        const pathCompare = PathUtils.compare(op1.get('path'), op2.get('path'));
+        if (pathCompare === 0) {
+            if (op1.get('offset') > op2.get('offset') && op1.get('offset') < op2.get('offset') + op2.get('length')) {
+                const mark = op2.get('mark');
+                let newMarks = op1.get('marks');
+                newMarks = newMarks.add({ type: mark.type, data: {} });
```

Comment:
Maybe we need to make it:
`newMarks = newMarks.add(Mark.create{ type: mark.type, data: {}});`